### PR TITLE
Example of creating new materials using settings API

### PR DESCRIPTION
**Pull request by ansjsia**
Added an example to the docs showing how to create new materials using the settings API, as an alternative to using the TUI.

### CODE REVIEW

**Review by hpohekar — APPROVED** [review]

File: doc/source/user_guide/materials.rst [hunk 1]
```diff
@@ -53,9 +53,34 @@ Use settings objects
...
     solver.setup.cell_zone_conditions.fluid["elbow-fluid"].material = "water-liquid"
+
+Create new material
+~~~~~~~~~~~~~~~~~~~
+
+**Python code**
+
+.. code:: python
+
+    solver.setup.materials.solid["mysolid"] = {}
```
Comment (seanpearsonuk) [inline doc/source/user_guide/materials.rst:73]: @ansjsia 
```
>>> mysolid = solver.setup.materials.solid.create("mysolid")
>>> mysolid.chemical_formula = "SiO2"
```
Note the above create syntax. Regardless of the means of creation, I prefer not to repeat the trainwreck over multiple lines. I am referring to duplication of solver.setup.materials.solid["mysolid"]. I know that this is what a lot of authors do, but this creates a negative impression, and we should change a lot of existing test and example code.

I tried also your dict style access. It only works for already existing materials. It does not create. It raises an exception if the material does not exist.

Comment (ansjsia) [reply doc/source/user_guide/materials.rst]: Thanks for the comments, I have revised the syntax in the following commit. Please let me know if further changes are required.

**Review by seanpearsonuk — APPROVED** [review]
Thanks @ansjsia that looks good now

**Review by mkundu1 — APPROVED** [review]

**Comment by h-krishnan** [comment]
@ansjsia @mkundu1 @seanpearsonuk Note that setting each property separately is a bit slower than setting them all at once via a dictionary.

**Pull request merged**